python/cudf/cudf/core/reshape.py

```diff
@@ -964,18 +969,48 @@ def unstack(df, level, fill_value=None):
     a
     1     5  <NA>     6  <NA>     7
     2  <NA>     8  <NA>     9  <NA>
+
+    Unstacking single level index dataframe:
+
+    >>> df = cudf.DataFrame({('c', 1): [1, 2, 3], ('c', 2):[9, 8, 7]})
+    >>> df.unstack()
+    c  1  0    1
+          1    2
+          2    3
+       2  0    9
+          1    8
+          2    7
+    dtype: int64
     """
+    if not isinstance(df, cudf.DataFrame):
+        raise ValueError("`df` should be a cudf Dataframe object.")
+
+    if df.empty:
+        raise ValueError("Cannot unstack an empty dataframe.")
+
     if fill_value is not None:
         raise NotImplementedError("fill_value is not supported.")
     if pd.api.types.is_list_like(level):
         if not level:
             return df
     df = df.copy(deep=False)
     if not isinstance(df.index, cudf.MultiIndex):
-        raise NotImplementedError(
-            "Calling unstack() on a DataFrame without a MultiIndex "
-            "is not supported"
-        )
+        if isinstance(df, cudf.DataFrame):
+            dtype = df._columns[0].dtype
+            if any(
```

Comment:
It won't matter much in most cases but I think it's slightly more efficient to use a single loop that raises `if df._columns[i].dtype != dtype` as opposed to creating a list with all the booleans and then doing an `any` reduction.

Reply:
Raised an issue since this comes up in a bunch of places https://github.com/rapidsai/cudf/issues/7067